templates/periodic.erb

```diff
@@ -46,3 +46,17 @@ APT::Periodic::Verbose "<%= @verbose %>";
 #      1:  progress report       (actually any string)
 #      2:  + command outputs     (remove -qq, remove 2>/dev/null, add -d)
 #      3:  + trace on
+<%- unless @random_sleep.nil? -%>
```

Comment:
I tested on a puppet4 & puppetserver environment and it works fine.

In this page says the following: "Puppet’s special undef value is roughly equivalent to nil in Ruby; it represents the absence of a value."

https://docs.puppetlabs.com/puppet/latest/reference/lang_data_undef.html